lib/codec_choose.c

```diff
@@ -84,57 +84,84 @@ BASE64_CODEC_FUNCS(sse41)
 BASE64_CODEC_FUNCS(sse42)
 BASE64_CODEC_FUNCS(avx)
 
+static bool avx2_supported(void);
+static bool avx_supported(void);
+static bool sse42_supported(void);
+static bool sse41_supported(void);
+static bool ssse3_supported(void);
```

Comment:
These forward declarations can be elided by defining the functions before first use.